### DISCUSSION TITLE
How can I create a "Forms" provider when using multiple forms?

### DISCUSSION THREAD

**Discussion by npearson72** [react-hook-form/react-hook-form#9344] [Q&A]
I'm doing something similar to the wizard example in the docs:

https://react-hook-form.com/advanced-usage/#WizardFormFunnel

I have multiple forms presented in an accordion UI. I'd like to create a provider (and custom hook) to access the `UseFormReturn` object, which includes the `formState` object that I need to use in different parts of my component tree.

But every time I try, for some reason my "native" (i.e. uncontrolled) validations stop working properly. Oddly though, my controlled validations seem to continue to work.

Here's what I've been doing:

```
export const FormsProvider = ({ children }) => {
  const stepOneForm = useForm({
    resolver: yupResolver(stepOneSchema),
    mode: 'onTouched'
  });

  const stepTwoForm = useForm({
    resolver: yupResolver(stepTwoSchema),
    mode: 'onTouched'
  });

  const value = useMemo(() => {
    return { stepOneForm, stepTwoForm };
  }, [stepOneForm, stepTwoForm]);

  return (
    <FormsContext.Provider value={value}>{children}</FormsContext.Provider>
  );
};
```

Things seem to work when I remove the `useMemo`, but I think using `useMemo` is the recommended way to create the `value` object for a provider.

Is there an example for doing this?

**Comment by leapful** [comment]
@npearson72 

Could you please provide a codesandbox to re-produce your problem ?
